app/helpers/components_helper.rb

```diff
@@ -0,0 +1,17 @@
+module ComponentsHelper
+  def self.register_components(*components, namespace: nil)
+    components.each do |component_name|
+      namespaced_component_name = [namespace, component_name].compact.join("_")
+      component_class_name = "#{[namespace, component_name].compact.join("/")}_component"
+
+      define_method(namespaced_component_name) do |*args, **kwargs, &block|
+        render component_class_name.classify.constantize.new(*args, **kwargs) do |component|
+          block.call(component) if block.present?
+        end
+      end
+    end
+  end
```

Comment:
I can see the appeal of this from an aesthetic point of view. However, I do wonder if it's trying to solve a problem that doesn't really exist?

Ideally we'd try to [avoid unnecessary metaprogramming](https://rubystyle.guide/#no-needless-metaprogramming), and I'm wondering if we could survive without this particular piece of metaprogramming.

This adds a layer of indirection that will presumably break the `Cmd + Click` _jump to definition_ feature in developer IDEs. And while it suggests a convention, it doesn't enforce it, so we end up with two different ways to render the same component.

I can certainly see the appeal of implementing a more 'Rails helper method'-like public API for the [govuk-components](https://govuk-components.netlify.app/) gem. But I'm wondering if it's really necessary within this project, where we plan on using ViewComponents internally within the app.

Reply:
I agree with Ollie here, selfishly perhaps, mostly because it breaks the `Cmd + Click` in my IDE. So I have to try to work out what the component is called based on the helper name and then find the definition separately. It's not such an easy developer experience.